docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+
+An object of `UserStatsCalculator` class is responsible for calculating the aggregate results from the user’s previous quiz results. For instance, its `calculateTotalPointsEarned()` method will iterate through the multiple topics stored in E-Duke-8 and calculate the total sum of the user’s past results of the quizzes done for those topics.
+
+On the other hand, an object of `TopicStatsCalculator` is used by the object of `Stats` class to calculate the topical results. In its constructor, the `TopicStatsCalculator` object uses the single `Topic` object passed into it to retrieve its specific `QuestionList` object. Thereafter, by iterating through the questions for the particular `QuestionList` object, the results for individual topics can be calculated with its methods.
+
+
+
+### 2.4. Storage Component
+
+#### 2.4.1. Design of TopicsStorage
+
+Given data for the topics and questions is loaded automatically from JSON (JavaScript Object Notation) files in the data folder. This is mainly facilitated through the `TopicsStorage` 
+class which handles accessing the file as well as converting from JSON into `Topic`, `Question` and `Option` objects. The class diagram below shows this relationship.
+
+![TopicsStorage Class Diagram](./images/TopicsStorage.png)
+
+The format of the JSON file is important as it is loaded in a particular way. This format was designed as an array of topics that hold the different properties for questions, options, hints and explanations. An example is as such:
+
+```json
+[
+  {
+    "topic": "OOP",
+    "questions": [
+      {
+        "description": "What is encapsulation?",
+        "hint": "check the textbook",
+        "explanation": "because option A is the best answer",
+        "options": [
+          {
+            "description": "A",
+            "correct": true
+          },
+          {
+            "description": "B",
+            "correct": false
+          },
+          {
+            "description": "C",
+            "correct": false
+          },
+          {
+            "description": "D",
+            "correct": false
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+#### 2.4.2. Implementation of TopicsStorage
+
+Given below is an example usage scenario of loading in two topics with two questions each.
+
+When the user launches the app, the main program will initialize a `TopicsStorage` object and call the `load` method 
+which will return a `TopicList` object. The following sequence diagram shows how the load operation works:
+
+![TopicsStorage::load Sequence Diagram](./images/TopicsStorage_load.png)
```

Comment:
![image](https://user-images.githubusercontent.com/47075579/97397494-51fd3d80-1924-11eb-8c46-482c64f1026a.png)
Could you consider making the initial arrow for the load() method longer? It might help to make it more visible and clearer

Reply:
I have made it longer as suggested, thank you for the advice.